### PR TITLE
Feature/reverse bnt

### PR DESCRIPTION
**Pull request by MathewWith**
Created reverse button

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -21,34 +23,64 @@ function App() {
     getCurrencies()
   }, [])
 
+  useEffect(() => {
+      const setValueLatestPrice = async () => {
```
Comment (CodingRuslan) [inline src/App.tsx]: Don't use value and price in one naming.
Price it's the wrong name for this variable, think about the more clear name

File: src/App.tsx [hunk 2]
```diff
@@ -21,34 +23,64 @@ function App() {
...
+    if(!isReversed){
+      setConvertedValue()
+    } 
+  }, [rightInputValue, leftCurrency])
 
   return (
-    <div className="App">
-      <h1>Change money</h1>
+    <div className="container">
+      <h1 className='container-title'>Change money</h1>
```
Comment (CodingRuslan) [inline src/App.tsx]: Use BEM

File: src/App.tsx [hunk 3]
```diff
@@ -21,34 +23,64 @@ function App() {
...
         allCurrencies={allCurrencies} 
         setLeftCurrency={setLeftCurrency} 
-        setRightCurrency={setRightCurrency}/>
-      <div>
-          <input type="text" onClick={() => setIsReversed(prev => !prev)} onChange={(e) => setLeftInputValue(e.target.value)}/>
-          <input type="text" onClick={() => setIsReversed(prev => !prev)} onChange={(e) => setRightInputValue(e.target.value)}/>
+        setRightCurrency={setRightCurrency}
+        leftCurrency={leftCurrency}
+        rightCurrency={rightCurrency}/>
+      <div className='container-inputs'>
```
Comment (CodingRuslan) [inline src/App.tsx]: BEM

File: src/App.tsx [hunk 4]
```diff
@@ -21,34 +23,64 @@ function App() {
...
+                 onChange={(e) => setLeftInputValue(e.target.value)}
+                 value={leftInputValue}
+          />
+          <input type="text" 
+                 onClick={() => setIsReversed(false)} 
+                 onChange={(e) => setRightInputValue(e.target.value)}
+                 value={rightInputValue}
+          />
       </div>
+      <button className='container-btn'
```
Comment (CodingRuslan) [inline src/App.tsx]: BEM

File: src/components/Select/Select.tsx [hunk 5]
```diff
@@ -1,14 +1,17 @@
 import { SelectProps } from "src/types/SelectTypes";
+import './Select.scss'
 
 export const Select = (props: SelectProps) => {
-    const {allCurrencies, setCurrency} = props;
+    const {allCurrencies, setCurrency, pickCurrency} = props;
     return (
-        <select onChange={(e) => setCurrency(e.target.value)}>
+        <select onChange={(e) => setCurrency(e.target.value)} value={pickCurrency} className='select-wrap'>
```
Comment (CodingRuslan) [inline src/components/Select/Select.tsx]: BEM

File: src/components/Select/Select.tsx [hunk 6]
```diff
@@ -1,14 +1,17 @@
 import { SelectProps } from "src/types/SelectTypes";
+import './Select.scss'
 
 export const Select = (props: SelectProps) => {
-    const {allCurrencies, setCurrency} = props;
+    const {allCurrencies, setCurrency, pickCurrency} = props;
```
Comment (CodingRuslan) [inline src/components/Select/Select.tsx]: pickCurrency - a bad name

File: src/shared/consts/api-key.ts [hunk 7]
```diff
@@ -1,2 +1,2 @@
 
-export const API_KEY = '50dbe660f41b73fe2f00a4af';
\ No newline at end of file
+export const API_KEY = '2bb2108fb19d88a23ddc5ece';
```
Comment (CodingRuslan) [inline src/shared/consts/api-key.ts]: use **dotenv ** library

File: src/types/IReverseButton.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+
+export interface IReverseButton {
```
Comment (CodingRuslan) [inline src/types/IReverseButton.ts]:
```suggestion
export interface ReverseButton {
```

File: src/types/IReverseButton.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (CodingRuslan) [inline src/types/IReverseButton.ts]: You created one file for one API request, now you create one file for one type. don't you see the addiction?

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/types/IReverseButton.ts (hunk 9 above)
Comment (MathewWith) [reply src/types/IReverseButton.ts]: Now see

File: src/App.tsx [hunk 10]
```diff
@@ -10,45 +11,90 @@ function App() {
...
-  const [leftCurrency, setLeftCurrency] = useState<string>('')
-  const [rightCurrency, setRightCurrency] = useState<string>('')
+  const [leftCurrency, setLeftCurrency] = useState<string>(allCurrencies[0])
+  const [rightCurrency, setRightCurrency] = useState<string>(allCurrencies[0])
+  const [exchangeRate, setExchangeRate] = useState<string|number>('')
+  const [currenciesSymbols, setCurrenciesSymbols] = useState<string[]>([])
 
   useEffect(() => {
-    const getCurrencies = async () => {
+    const setSymbols = async () => {
```
Comment (CodingRuslan) [inline src/App.tsx]: Why do you create separate useEffects for API request(GET)?

File: src/service/CurrencyService.ts [hunk 11]
```diff
@@ -1,26 +1,43 @@
 import axios from "axios"
-import {API_KEY} from "src/shared/consts/api-key"
+
+export const getCurrenciesSymbols = async (): Promise<string[]> => {
+    let response = await axios.get(`https://gist.githubusercontent.com/nhalstead/4c1652563dd13357ab936fc97703c019/raw/d5de097ef68f37501fb4d06030ca49f10f5f963a/currency-symbols.json`)
+    console.log(response.data)
```
Comment (CodingRuslan) [inline src/service/CurrencyService.ts]: remove console.log

File: src/service/CurrencyService.ts [hunk 12]
```diff
@@ -1,26 +1,43 @@
...
+    return response.data
+}
+
+export const getExchangeRate = async (leftCurrency: string, rightCurrency: string): Promise<number|string> => {
+    if(leftCurrency === '') {
+        return ''
+    }
+    let response = await axios.get(
+        `https://fcsapi.com/api-v3/forex/latest?symbol=${leftCurrency}/${rightCurrency}&access_key=W5ndOKH68FvIDvLZJR4FQXs`)
+    console.log(response.data.response[0].ch);
```
Comment (CodingRuslan) [inline src/service/CurrencyService.ts]: Remove console log

File: src/service/CurrencyService.ts [hunk 13]
```diff
@@ -1,26 +1,43 @@
...
    if(inputValue === ''){
        return ''
    }
-   
+    
     const response = await axios.get(
-        `https://v6.exchangerate-api.com/v6/${API_KEY}
-        /pair/${inputCurrency}/${outputCurrency}/${inputValue}`)
+        `https://v6.exchangerate-api.com/v6/${process.env.REACT_APP_API_KEY}/pair/${inputCurrency}/${outputCurrency}/${inputValue}`)
+        console.log(response.data.conversion_result);
```
Comment (CodingRuslan) [inline src/service/CurrencyService.ts]: remove console.log

File: src/types/types.ts [hunk 14]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (CodingRuslan) [inline src/types/types.ts]: types/types.ts ?

Comment (CodingRuslan) [reply src/types/types.ts]: Why you don't name your service like **service/service.ts** ?

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/App.scss [hunk 15]
```diff
@@ -1,38 +1,38 @@
...
-}
-
-.App-link {
-  color: #61dafb;
-}
-
-@keyframes App-logo-spin {
-  from {
-    transform: rotate(0deg);
+  .container__inputs {
```
Comment (CodingRuslan) [inline src/App.scss]: container_**inputs** - bad name

File: src/App.scss [hunk 16]
```diff
@@ -1,38 +1,38 @@
...
+    display: flex;
+    justify-content: space-between;
+    padding: 3% 30%;
+    height: 17%;
+    input {
```
Comment (CodingRuslan) [inline src/App.scss]: You shouldn't use pure tags, add valid class for input and use in scss

File: src/App.scss [hunk 17]
```diff
@@ -1,38 +1,38 @@
...
+      font-size: 40px;
+      border: none;
+      outline: none;
+      border-bottom: 1px solid silver;
+      font-weight: 100;
+    }
   }
-  to {
-    transform: rotate(360deg);
+  .container__wrapper {
```
Comment (CodingRuslan) [inline src/App.scss]: Container == wrapper. You should use one of them

File: src/App.scss [hunk 18]
```diff
@@ -1,38 +1,38 @@
...
+    display: flex;
+    justify-content: space-between;
+    padding: 0% 30%;
+    .container__btn {
```
Comment (CodingRuslan) [inline src/App.scss]: If you write nested classes you should use previous states, like as container__wrapper--btn

File: src/App.tsx (hunk 10 above)
Comment (CodingRuslan) [reply src/App.tsx]: My questions not changed

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

**Comment by CodingRuslan** [comment]
Fix conflicts

File: src/App.tsx (hunk 1 above)
Comment (MathewWith) [reply src/App.tsx]: +

File: src/App.tsx (hunk 2 above)
Comment (MathewWith) [reply src/App.tsx]: +

File: src/App.tsx (hunk 3 above)
Comment (MathewWith) [reply src/App.tsx]: +

File: src/App.tsx (hunk 4 above)
Comment (MathewWith) [reply src/App.tsx]: +

File: src/components/Select/Select.tsx (hunk 5 above)
Comment (MathewWith) [reply src/components/Select/Select.tsx]: +

File: src/components/Select/Select.tsx (hunk 6 above)
Comment (MathewWith) [reply src/components/Select/Select.tsx]: +

File: src/App.scss (hunk 15 above)
Comment (MathewWith) [reply src/App.scss]: I got it, will read BEM agy

File: src/App.scss (hunk 16 above)
Comment (MathewWith) [reply src/App.scss]: Okay

File: src/App.scss (hunk 17 above)
Comment (MathewWith) [reply src/App.scss]: +

File: src/App.scss (hunk 18 above)
Comment (MathewWith) [reply src/App.scss]: +

File: src/App.tsx (hunk 10 above)
Comment (MathewWith) [reply src/App.tsx]: My mistake, fixed it

File: src/service/CurrencyService.ts (hunk 11 above)
Comment (MathewWith) [reply src/service/CurrencyService.ts]: Okay

File: src/service/CurrencyService.ts (hunk 13 above)
Comment (MathewWith) [reply src/service/CurrencyService.ts]: Okay

File: src/types/types.ts (hunk 14 above)
Comment (MathewWith) [reply src/types/types.ts]: I see that I have a folder “service”, and I thought that this name not free 
I’ll fixed it

Comment (MathewWith) [reply src/types/types.ts]: fixe*

Comment (MathewWith) [reply src/types/types.ts]: Aaa, I misunderstood
But I got it now

File: src/service/CurrencyService.ts (hunk 12 above)
Comment (MathewWith) [reply src/service/CurrencyService.ts]: Done

File: src/App.scss (hunk 18 above)
Comment (MathewWith) [reply src/App.scss]: We talked about it

File: src/App.scss [hunk 19]
```diff
@@ -1,38 +1,38 @@
...
-
-.App-logo {
-  height: 40vmin;
-  pointer-events: none;
-}
-
-@media (prefers-reduced-motion: no-preference) {
-  .App-logo {
-    animation: App-logo-spin infinite 20s linear;
+  .container__title {
```
Comment (CodingRuslan) [inline src/App.scss]:
```suggestion
 &__title {
```

File: src/App.tsx [hunk 20]
```diff
@@ -10,44 +11,88 @@ function App() {
...
```
Comment (CodingRuslan) [inline src/App.tsx]: And now try to join two functions for getting a date to one

File: src/components/ListCurrencies/ListCurrencies.tsx [hunk 21]
```diff
@@ -1,18 +1,15 @@
...
 
-interface IListCurrencies {
-    allCurrencies: string[],
-    setLeftCurrency: React.Dispatch<React.SetStateAction<string>>,
-    setRightCurrency: React.Dispatch<React.SetStateAction<string>>
-}
 
-export const ListCurrencies = (props: IListCurrencies ) => {
-    const {allCurrencies, setLeftCurrency, setRightCurrency} = props 
+export const ListCurrencies = (props: ListCurrenciesProps ) => {
```
Comment (CodingRuslan) [inline src/components/ListCurrencies/ListCurrencies.tsx]: Destructurization

File: src/types/FunctionProps.ts [hunk 22]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (CodingRuslan) [inline src/types/FunctionProps.ts]: FunctionProps ? What is function?

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

**Comment by CodingRuslan** [comment]
Fix conflicts

File: src/types/FunctionProps.ts (hunk 22 above)
Comment (MathewWith) [reply src/types/FunctionProps.ts]: Sorry 
Props of functions - it's more correct

File: src/App.tsx (hunk 20 above)
Comment (MathewWith) [reply src/App.tsx]: Done

File: src/components/ListCurrencies/ListCurrencies.tsx (hunk 21 above)
Comment (MathewWith) [reply src/components/ListCurrencies/ListCurrencies.tsx]: Done

File: src/App.tsx [hunk 23]
```diff
@@ -12,50 +14,93 @@ function App() {
   const [isReversed, setIsReversed] = useState<boolean>(false)
   const [leftCurrency, setLeftCurrency] = useState<string>('')
   const [rightCurrency, setRightCurrency] = useState<string>('')
+  const [exchangeRate, setExchangeRate] = useState<string|number>('')
+  const [allCurrenciesSymbols, setAllCurrenciesSymbols] = useState<Symbol[]>([])
+  const [leftSymbol, setLeftSymbol] = useState<string|undefined>()
+  const [rightSymbol, setRightSymbol] = useState<string|undefined>()
+
+  useEffect(() => {
+    addSymbolToInput({leftCurrency, rightCurrency, allCurrenciesSymbols, setLeftSymbol, setRightSymbol})
```
Comment (CodingRuslan) [inline src/App.tsx]: First of all: don't add new features to the old rep/functional
second: your useEffect will run when you change the left or right  currency input

File: src/App.tsx [hunk 24]
```diff
@@ -12,50 +14,93 @@ function App() {
...
+  },[allCurrenciesSymbols, leftCurrency, rightCurrency])
 
   useEffect(() => {
-    const getCurrencies = async () => {
+    const setSymbolsAndAllCurrencies = async () => {
+      const response = await getCurrenciesSymbols()
```
Comment (CodingRuslan) [inline src/App.tsx]: from getCurrenciesSymbols you get response 
from getAllCurrencies you don't get response ?

File: src/App.tsx [hunk 25]
```diff
@@ -12,50 +14,93 @@ function App() {
...
+  const [leftSymbol, setLeftSymbol] = useState<string|undefined>()
+  const [rightSymbol, setRightSymbol] = useState<string|undefined>()
+
+  useEffect(() => {
+    addSymbolToInput({leftCurrency, rightCurrency, allCurrenciesSymbols, setLeftSymbol, setRightSymbol})
+  },[allCurrenciesSymbols, leftCurrency, rightCurrency])
 
   useEffect(() => {
-    const getCurrencies = async () => {
+    const setSymbolsAndAllCurrencies = async () => {
```
Comment (CodingRuslan) [inline src/App.tsx]: if we will add a new API request(get a user for example), how you will name this function?

File: src/App.tsx [hunk 26]
```diff
@@ -12,50 +14,93 @@ function App() {
...
-          <input 
-              type="text" 
-              onClick={() => setIsReversed(true)} 
-              onChange={(e) => setLeftInputValue(+e.target.value)}/>
-          <input 
-              type="text" 
-              onClick={() => setIsReversed(false)} 
-              onChange={(e) => setRightInputValue(+e.target.value)}/>
+        rightCurrency={rightCurrency}/>
+      <div className='container__inputs'>
```
Comment (CodingRuslan) [inline src/App.tsx:70]: Omg why are you adding new features to old PR? Do you want infinity PR?

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 26 above)
Comment (MathewWith) [reply src/App.tsx:70]: It's really not good
I tried to fix the conflicts and did it wrong
When I saw it, I was disappointed

File: src/App.tsx (hunk 24 above)
Comment (MathewWith) [reply src/App.tsx]: Fixed it

File: src/App.tsx (hunk 25 above)
Comment (MathewWith) [reply src/App.tsx]: apiRequests or gettingDate

File: src/App.tsx (hunk 23 above)
Comment (MathewWith) [reply src/App.tsx]: +

File: src/App.tsx (hunk 25 above)
Comment (MathewWith) [reply src/App.tsx]: gettingData*

File: src/App.scss [hunk 27]
```diff
@@ -1,38 +1,47 @@
...
+      font-size: 30px;
+    }
   }
-  to {
-    transform: rotate(360deg);
+  &__content {
+    display: flex;
+    justify-content: space-between;
+    padding: 0% 30%;
+    .container__content--btn {
```
Comment (CodingRuslan) [inline src/App.scss]: Why you don't use & ?

File: src/App.tsx [hunk 28]
```diff
@@ -12,50 +14,90 @@ function App() {
...
+  const [leftSymbol, setLeftSymbol] = useState<string|undefined>()
+  const [rightSymbol, setRightSymbol] = useState<string|undefined>()
 
   useEffect(() => {
-    const getCurrencies = async () => {
-      const currencies = await getAllCurrencies()
-      setAllCurrencies(currencies)
+    const gettingData = async () => {
+      const responseOfSymbols = await getCurrenciesSymbols()
+      const responseOfCurrencies = await getAllCurrencies()
```
Comment (CodingRuslan) [inline src/App.tsx]:
```suggestion
      const symbols = await getCurrenciesSymbols()
      const currencies = await getAllCurrencies()
```

File: src/types/PropsOfFunctions.ts [hunk 29]
```diff
@@ -0,0 +1,45 @@
+
```
Comment (CodingRuslan) [inline src/types/PropsOfFunctions.ts]: file name should be CurrencyTypes

**Review by CodingRuslan — CHANGES_REQUESTED** [review]

File: src/types/PropsOfFunctions.ts (hunk 29 above)
Comment (MathewWith) [reply src/types/PropsOfFunctions.ts]: Okay
I think that I understood now

File: src/App.tsx (hunk 28 above)
Comment (MathewWith) [reply src/App.tsx]: I got it
I had to understood it myself

File: src/App.scss (hunk 27 above)
Comment (MathewWith) [reply src/App.scss]: Ok, I can say you that I used it wrong before and I was wrong 
Now I understand how to use it

**Pull request merged**